Class 06

robert-hatfield opened the pull request:
This is a partial implementation of loading data w/ AJAX from JSON. I've spent about 2 - 3 hours so far on integrating techniques covered in classes 5 & 6.
    
All data objects are saving to local storage, so they're definitely loading from the .json file. But there's nothing in the `Project.all` array. I need to figure out why only a single blank project article
is being rendered.

**Next step** - Check order of operations:
  1. Loading
  2. Storing to array
  3. Building articles from template
  4. Appending to DOM

robert-hatfield committed:
Update TODO document with requirements for class 6

I've also consolidated previous TODO documents into a single document.
 Using separate docs was getting unmanageable.

robert-hatfield committed:
Move data storage to JSON file

robert-hatfield committed:
Move variables and functions off global scope

robert-hatfield committed:
Partially implement loading data w/ AJAX from JSON

All data objects are saving to local storage, so they're definitely
 loading from the .json file. But there's nothing in the Project.all
  array. I need to figure out why only a single blank project article
  is being rendered.
**Next step** - Check order of operations:
 - [ ] Loading
 - [ ] Storing to array
 - [ ] Building articles from template
 - [ ] Appending to DOM

elaewin commented on scripts/app.js:
```diff
@@ -11,16 +9,28 @@ function Project(input) {
-rawProjectData.forEach(function(rawObject) { // Pass all raw data object literals to constructor, and add the resulting object article to the projects array
-  projects.push(new Project(rawObject))
-});
+Project.loadAll = function (rawObject) {
```
This method is what is causing your loading problems. (It's missing something.)

robert-hatfield committed:
Solve loading of data w/ AJAX from JSON

The array at Project.all was not being modified for every object in our
 JSON object. Solved the .forEach() method. This ensures all 4 **raw**
 objects are passed into the constructor, and the resulting
 **constructed** objects are pushed into Project.all.
I renamed some of the parameters to more clearly identify what was being
 passed into each of these methods.

robert-hatfield commented:
This is fixed. Moved `.push()` into a `.forEach()` to ensure all rawObjects run through the constructor **and** the results are pushed into the array.
